state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1201,199 +1226,261 @@ pub mod pallet {
-					swap_request_id,
-					swap_id: swap.swap_id(),
-					// To be consistent with `swap_output` and `intermediate_amount` (which do
-					// not include the network fee), we report input amount without the network fee
-					// for swaps from STABLE_ASSET:
-					input_amount: if swap.input_asset() == STABLE_ASSET {
-						swap.stable_amount.unwrap_or_else(|| {
-							log_or_panic!("stable amount must be set for swaps from STABLE_ASSET");
-							swap.input_amount()
-						})
-					} else {
-						swap.input_amount()
-					},
-					input_asset: swap.input_asset(),
-					network_fee: swap.network_fee_taken.unwrap_or_default(),
-					output_asset: swap.output_asset(),
-					output_amount,
-					intermediate_amount: swap.intermediate_amount(),
-				});
+			Self::deposit_event(Event::<T>::SwapRequestCompleted { swap_request_id: request.id });
 
-				match &mut request.state {
-					SwapRequestState::Ccm { state, ccm_deposit_metadata, output_address } =>
-						match state {
-							CcmSwapState::PrincipalSwapScheduled {
-								principal_swap_id,
-								gas_budget,
-								other_gas_asset,
-							} => {
-								debug_assert!(swap.swap_id() == *principal_swap_id);
-
-								if let Some(other_gas_asset) = other_gas_asset {
-									*state = Self::schedule_ccm_gas_swap(
-										swap_request_id,
-										request.input_asset,
-										*other_gas_asset,
-										*gas_budget,
-										output_amount,
-									);
+			match request.state {
+				SwapRequestState::Regular {
+					dca_state: DcaState { remaining_input_amount, accumulated_output_amount, .. },
+					output_address,
+				} => {
+					// Refund the failed swap and any unused input amount:
+					Self::egress_for_swap(
+						request.id,
+						swap.input_amount + remaining_input_amount,
+						request.input_asset,
+						refund_params.refund_address,
+						None, /* ccm */
+						true, /* refund */
+					);
 
-									// We are not done with the request, so put it back
-									SwapRequests::<T>::insert(swap_request_id, request);
-								} else {
-									Self::process_ccm_outcome(
-										swap_request_id,
-										request.output_asset,
-										output_address.clone(),
-										output_amount,
-										*gas_budget,
-										ccm_deposit_metadata.clone(),
-									);
-								}
-							},
-							CcmSwapState::GasSwapScheduled {
-								gas_swap_id,
-								principal_output_amount,
-							} => {
-								debug_assert!(swap.swap_id() == *gas_swap_id);
+					// In case of DCA we may have partially swapped and now have some output asset
+					// to egress to the output address:
+					if accumulated_output_amount > 0 {
+						Self::egress_for_swap(
+							swap.swap_request_id,
+							accumulated_output_amount,
+							request.output_asset,
+							output_address,
+							None,  /* ccm */
+							false, /* refund */
+						);
+					}
+				},
+				SwapRequestState::Ccm { output_address: _, .. } => {
+					// TODO: add remaining DCA input to the refunded amount:
```

Comment:
This TODO should be included in this PR? Or a future one?

Reply:
This is just a reminder for a future PR which will add support for FoK to CCM swaps.